src/ray/core_worker/reference_count.h

```diff
@@ -563,21 +610,51 @@ class ReferenceCounter : public ReferenceCounterInterface,
       }
     }
 
+    /// Access BorrowInfo without modifications.
+    /// Returns the default value of the struct if it is not set.
+    const BorrowInfo &borrow() const {
+      if (borrow_info == nullptr) {
+        static auto *default_info = new BorrowInfo();
```

Comment:
Thanks for the suggestion. Using `static local` raw pointers to avoid the objects being destructed was a common idiom in my previous company, so people need not think about the implication of destroying `default_info` on shutdown. But I can update this if it conflicts with Ray's style.